### PR TITLE
[nl] Update ranking calculation for BLS data

### PR DESCRIPTION
**Pull request by chejennifer**
for BLS variables, when fetching data to calculate time delta across places, fetch data from a facet that is not P1M because that is too much data and will cause mixer to error when fetching it for US counties

Screenshot 2024-02-13 at 9 21 57 AM

### CODE REVIEW

File: server/lib/nl/common/rank_utils.py [hunk 1]
```diff
@@ -55,6 +55,157 @@
     (types.TimeDeltaType.CHANGE, types.RankingType.LOW):
         False,
 }
+# Set of stat vars that need to check which facet is being used before doing
+# ranking because they have facets that are too fine in granularity.
+# This list of svs are svs in the medium embedding index that have a BLS source.
+SVS_TO_CHECK_FACET = set([
```
Comment (pradh) [inline server/lib/nl/common/rank_utils.py:61]: Maybe move this to a separate file?

File: server/lib/nl/config_builder/timeline.py [hunk 2]
```diff
@@ -78,8 +78,7 @@ def ranked_timeline_collection_block(builder: base.Builder,
       else:
         chart_title = base.decorate_chart_title(title=sv2thing.name[sv_dcid],
                                                 place=place)
-      facet_id = get_facet_id(sv_dcid, single_date, cspec.sv_place_facet_id,
```
Comment (pradh) [inline server/lib/nl/config_builder/timeline.py:81]: Why is this removed?

**Review by pradh — COMMENTED** [review]
Nice!

File: server/lib/nl/config_builder/timeline.py (hunk 2 above)
Comment (chejennifer) [reply server/lib/nl/config_builder/timeline.py:81]: it isn't removed, the function is just updated to not take date as a param anymore. `get_facet_id` used to take a `date` as a param and only return an id if there is a date passed in. We now want to get facet id even in non date situations, so I updated the function to get facet id if sv_place_facet_id is not empty.

File: server/lib/nl/common/rank_utils.py (hunk 1 above)
Comment (chejennifer) [reply server/lib/nl/common/rank_utils.py]: saw that there was a constants file, moved it there

File: server/lib/nl/common/constants.py [hunk 3]
```diff
@@ -845,3 +845,155 @@
 ROOT_TOPIC = 'dc/topic/Root'
 
 PROJECTED_TEMP_TOPIC = 'dc/topic/ProjectedClimateExtremes'
+
+# Set of stat vars that need to check which facet is being used before doing
+# ranking because they have facets that are too fine in granularity.
+# This list of svs are svs in the medium embedding index that have a BLS source.
+SVS_TO_CHECK_FACET = set([
```
Comment (pradh) [inline server/lib/nl/common/constants.py:852]: nit: frozenset

File: server/lib/nl/common/constants.py [hunk 4]
```diff
@@ -845,3 +845,155 @@
 ROOT_TOPIC = 'dc/topic/Root'
 
 PROJECTED_TEMP_TOPIC = 'dc/topic/ProjectedClimateExtremes'
+
+# Set of stat vars that need to check which facet is being used before doing
+# ranking because they have facets that are too fine in granularity.
+# This list of svs are svs in the medium embedding index that have a BLS source.
```
Comment (pradh) [inline server/lib/nl/common/constants.py:851]: Add a comment how you got this list?  Maybe link to query

File: server/lib/nl/common/rank_utils.py [hunk 5]
```diff
@@ -87,9 +87,21 @@ def rank_places_by_series_growth(
...
+    series_facets = fetch.series_facet(entities=places,
+                                       variables=[sv],
+                                       all_facets=True)
+    for facet_id, facet_metadata in series_facets.get('facets', {}).items():
+      # Choose any facet that is not P1M
+      if facet_metadata and facet_metadata.get('observationPeriod',
+                                               '') != 'P1M':
+        facet_to_fetch = [facet_id]
+        break
   series_data = fetch.series_core(entities=places,
```
Comment (pradh) [inline server/lib/nl/common/rank_utils.py:101]: Order P1Y > P3M > P1M ?

File: server/lib/nl/config_builder/timeline.py (hunk 2 above)
Comment (pradh) [reply server/lib/nl/config_builder/timeline.py:81]: Got it.

**Review by pradh — APPROVED** [review]
Thanks for the fix!

File: server/lib/nl/common/rank_utils.py (hunk 5 above)
Comment (chejennifer) [reply server/lib/nl/common/rank_utils.py:110]: done

File: server/lib/nl/common/constants.py (hunk 4 above)
Comment (chejennifer) [reply server/lib/nl/common/constants.py:851]: done

File: server/lib/nl/common/constants.py (hunk 3 above)
Comment (chejennifer) [reply server/lib/nl/common/constants.py]: done

**Review by pradh — APPROVED** [review]
Thanks for the changes!

File: server/lib/nl/common/constants.py [hunk 6]
```diff
@@ -849,7 +849,8 @@
 # Set of stat vars that need to check which facet is being used before doing
 # ranking because they have facets that are too fine in granularity.
 # This list of svs are svs in the medium embedding index that have a BLS source.
-SVS_TO_CHECK_FACET = set([
+# Generated using: https://console.cloud.google.com/bigquery?sq=429015563165:7414cf70660c4d7fbf0731149b2cd7a7
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply server/lib/nl/common/constants.py]: done

**Comment by chejennifer** [comment]
thanks for the review!
